plugins/aea-ledger-hwi/aea_ledger_hwi/account.py

```diff
@@ -0,0 +1,356 @@
+    ) -> SignedMessage:
+        """Sign a EIP191 message"""
+
+        raise NotImplementedError()
+
+    @staticmethod
+    def encode_transaction(
+        transaction: TypedTransaction,
+        is_eip1559_tx: bool = False,
+        key_index: Optional[int] = None,
+    ) -> bytes:
+        """Build and encode transaction"""
+
+        # BIP32 Path m/44'/cointype'/account'/change/address
+        path = m / h(44) / h(60) / h(key_index or 0) / 0 / 0
+        path_construct = PrefixedArray(Byte, Int32ub)
+        path_apdu = path_construct.build(path.to_list())
+
+        if is_eip1559_tx:
+            as_dict = transaction.as_dict()
+            tx = UnsignedDynamicTransaction.from_dict(
+                field_dict=dict(
+                    chain_id=as_dict["chainId"],
+                    nonce=as_dict["nonce"],
+                    max_priority_fee_per_gas=as_dict["maxPriorityFeePerGas"],
+                    max_fee_per_gas=as_dict["maxFeePerGas"],
+                    gas_limit=as_dict.get("gas", 21000),
+                    destination=as_dict["to"],
+                    amount=as_dict["value"],
+                    data=as_dict["data"],
+                    access_list=as_dict.get("accessList", []),
+                )
+            )
+            return (
+                path_apdu
+                + transaction.transaction_type.to_bytes(1, "big")
+                + rlp.encode(tx)
+            )
+
+        as_dict = transaction.as_dict()
+        tx = UnsignedTransaction.from_dict(
+            field_dict=dict(
+                nonce=as_dict["nonce"],
+                gasPrice=as_dict["gasPrice"],
+                gas=as_dict["gas"],
+                to=as_dict["to"],
+                value=as_dict["value"],
+                data=as_dict["data"],
+            )
+        )
+        return path_apdu + rlp.encode(tx)
+
+    def _sign_transaction_on_hwi(
+        self,
+        unsigned_transaction: TypedTransaction,
+        key_index: Optional[int] = None,
+        device_index: Optional[int] = None,
+        chain_id: int = 1,
+    ) -> HWISignedTransaction:
+        """Hardware wallet signed transaction"""
+
+        ins = 0x04
+        p2 = 0x00
+
+        client = self.get_client(
+            device_index=device_index,
+        )
+
+        is_eip1559_tx = (
+            hasattr(unsigned_transaction, "transaction_type")
+            and unsigned_transaction.transaction_type == 2
+        )
+
+        request_data = self.encode_transaction(
+            transaction=unsigned_transaction,
+            is_eip1559_tx=is_eip1559_tx,
+            key_index=key_index,
+        )
+        raw_response = bytes()
+        for idx, each in enumerate(chunk(request_data, 255)):
+            p1 = 0x00 if idx == 0 else 0x80
+            raw_response = client.apdu_exchange(ins=ins, data=each, p1=p1, p2=p2)
+
+        parsed_response = Struct(
+            v=BytesInteger(1),
+            r=BytesInteger(32),
+            s=BytesInteger(32),
```

Comment:
These magic values are best declared as constants. This applies throughout. Someone else will struggle to understand what these constants represent otherwise.